packages/core/src/app/customer/Customer.tsx

```diff
@@ -148,10 +157,43 @@ class Customer extends Component<CustomerProps & WithCheckoutCustomerProps, Cust
             privacyPolicyUrl,
             providerWithCustomCheckout,
             requiresMarketingConsent,
+            isStripeLinkEnabled,
             onUnhandledError = noop,
+            step,
+            updateStripeLinkAuthenticated,
+
         } = this.props;
 
         return (
+            isStripeLinkEnabled ?
+                <StripeGuestForm
+                    canSubscribe={ canSubscribe }
+                    checkoutButtons={
+                        <CheckoutButtonList
```

Comment:
Can you explain why do we put `<CheckoutButtonList />` inside `<StripeGuestForm/>`?
Also, can we make stripe distinction in the `<GuestForm/>`?

Reply:
Sure @bc-peng, I'm using CheckoutButtonList because `StripeLink` should have the same behavior that BC users are used to, which includes showing buttons for `Apple pay`, `Google pay`, `Paypal`, etc... and the difference between `guest form` and `stripe guest form` is:

Instead of loading a component for the email, we create a `div` in which the `stripe component for the customer` will be inserted.

We use extra information from the `stripe iframe`, it provides us with additional information when it's a stripe user or not, it makes a challenge for login and has its own logout, the logic of the button also has been changed for example:

If you are a detected `Stripe user`, you will automatically go to the next step without clicking (could be shipping)

the button is shown if it is a `BC guest`, if it is a stripe customer it is hidden, and if you are a `stripe customer` and decided to return to the `customer step` the button will be displayed with a specific text in case you do not want to make any change in the mail.

I hope the videos above can better explain the behavior, if not, please let me know and I will record a video specifically, explaining the customer's step

As a resume: We are using `CheckoutButtonList` inside `StripeGuestForm` because we want to show `wallets` such as `Apple pay`, `Google pay`, `Amazon pay`, etc...
And we made a copy of `GuestForm` because we have a very specific logic only for `StripeLink`, and we don't want to affect the `guest customer` just in case we have to make any other change

Reply:
Hi Alan, in this case, it is logical to make`<GuestForm />` flexible enough to accept this kind of customization like `StripeLink`. Can you consider creating a `prop` or using `children`?